crates/matrix-sdk/src/account.rs

```diff
@@ -980,6 +987,131 @@ impl Account {
             .await?;
         Ok(())
     }
+
+    /// Observes the media preview configuration.
+    ///
+    /// This value is linked to the [MSC 4278](https://github.com/matrix-org/matrix-spec-proposals/pull/4278) which is still in an unstable state.
+    ///
+    /// This will return the initial value of the configuration and a stream
+    /// that will yield new values as they are received.
+    ///
+    /// The initial value is the one that was stored in the account data
+    /// when the client was started. If no value was found in either the stable
+    /// and unstable type, the default configuration will be returned.
+    /// and the following code is using a temporary solution until we know which
+    /// Matrix version will support the stable type.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// # use matrix_sdk::Client;
+    /// # use matrix_sdk::ruma::events::media_preview_config::MediaPreviews;
+    /// # use url::Url;
+    /// # async {
+    /// # let homeserver = Url::parse("http://localhost:8080")?;
+    /// # let client = Client::new(homeserver).await?;
+    /// let account = client.account();
+    ///
+    /// let (initial_config, mut config_stream) =
+    ///     account.observe_media_preview_config().await?;
+    ///
+    /// println!("Initial media preview config: {:?}", initial_config);
+    ///
+    /// while let Some(new_config) = config_stream.next().await {
+    ///     println!("Updated media preview config: {:?}", new_config);
+    /// }
+    /// # anyhow::Ok(()) };
+    /// ```
+    pub async fn observe_media_preview_config(
+        &self,
+    ) -> Result<
+        (MediaPreviewConfigEventContent, impl Stream<Item = MediaPreviewConfigEventContent>),
+        Error,
+    > {
+        let initial_value = self.get_media_preview_config_event_content().await?;
+
+        // We need to create two observers, one for the stable event and one for the
+        // unstable and combine them into a single stream.
+        let first_observer = self
+            .client
+            .observe_events::<GlobalAccountDataEvent<MediaPreviewConfigEventContent>, ()>();
+
+        let stream = first_observer.subscribe().map(|event| event.0.content);
+
+        let second_observer = self
+            .client
+            .observe_events::<GlobalAccountDataEvent<UnstableMediaPreviewConfigEventContent>, ()>();
+
+        let second_stream = second_observer.subscribe().map(|event| event.0.content.0);
+
```

Comment:
I just realized, we should probably first create the streams, then fetch the initial value, as we'd otherwise will have a race where we'll miss an update that happens while we fetch the initial value.

So let's fix the order here.